### DISCUSSION TITLE
How to ignore routes and filter GraphQL URLs?

### DISCUSSION THREAD

**Discussion by amaleszewski** [ThreeMammals/Ocelot#1879] [Q&A] [answered]
## Expected Behavior / New Feature

Hello! I want to ignore some routes for the Ocelot.

## Actual Behavior / Motivation for New Feature

I have a on my gateway GraphQL Stitching already, which is under `https://localhost:5001/api/graphql`. Without Ocelot I can make a POST request to that URL and it's redirecting my request to proper microservices, but after adding Ocelot and configuring rest of my routes which are REST, this GraphQL endpoint doesn't work anymore but of course routes configured in Ocelot works. I tried add the endpoints with the same downstream/upstream paths but it doesn't work.

Solution which I made, but I feel that it should be done better:

![image](https://user-images.githubusercontent.com/22931942/148246068-37c6a9d4-5a3a-4407-a3e5-66fd9f2cd508.png)

## Specifications

  - Version: 17.0.1
  - Platform: .NET 6
  - Subsystem: macOS

**Comment by raman-m** [comment, marked as answer]
Hi Adrian!
Sorry for the late reply!

You don't need to filter GraphQL URLs or ignore routes. Now OData, GraphQL URLs should be routed through Ocelot without the problems because OData and GraphQL technology use query string special syntax to send requests.
In [22.0.0](https://github.com/ThreeMammals/Ocelot/releases/tag/22.0.0) release we've fixed the OData URLs routing issues. See more in:
- commit https://github.com/ThreeMammals/Ocelot/commit/ae43f32fe32b322487623e4b1a22299c3d91bb73
- PR #1182 
- Issue #1174

We have special unit/acceptance tests which do the testing of URLs like OData. I guess GraphQL URLs have just different query parameter name.

Let us know testing results after updating to v22.0 plz!
